pkgs/tools/system/tiptop-py/default.nix

```
@@ -0,0 +1,46 @@
+{ lib
+, fetchFromGitHub
+, python3
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "tiptop-py";
+  version = "0.2.8";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "nschloe";
+    repo = "tiptop";
+    rev = "v${version}";
+    hash = "sha256-3nrYwcdqGGRvYB5FOW/OCqlRHIlCdkxClBfyKnVC1rQ=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    setuptools
+    py-cpuinfo
+    distro
+    psutil
+    rich
+    textual
+  ];
+
+  checkInputs = [ python3.pkgs.pytest ];
+
+  checkPhase = ''
+    pytest
+  '';
+
+  pythonImportsCheck = [
+    "tiptop"
+  ];
+
+  meta = with lib; {
+    description = "Command-line system monitoring tool in the spirit of top";
+    homepage = "https://github.com/nschloe/tiptop";
+    license = licenses.mit;
+    # Use a higher priority to shadow `tiptop` package.
+    priority = 1;
```

```suggestion
    priority = 5;
```
We should never use the highest priority by default.

---

Failed when set to `5`
```
error: builder for '/nix/store/pyvmprs4p4ak5ab17159hn1h5dwbbznz-home-manager-path.drv' failed with exit code 25;
       last 1 log lines:
       > error: collision between `/nix/store/c7nxr4smp5k6ykksa5dd45l9ayfsmvig-tiptop-py-0.2.8/bin/tiptop' and `/nix/store/29d07n4g2w12mj1n37nnl5b6298pz86n-tiptop-2.3.1/bin/tiptop'
```